constraint/pkg/client/client.go

```diff
@@ -189,22 +264,33 @@ func (c *Client) RemoveTemplate(ctx context.Context, templ *templates.Constraint
 	c.mtx.Lock()
 	defer c.mtx.Unlock()
 
-	err := c.driver.RemoveTemplate(ctx, templ)
-	if err != nil {
-		return resp, err
-	}
-
 	name := templ.GetName()
 
-	template, found := c.templates[name]
-
+	cached, found := c.templates[name]
 	if !found {
 		return resp, nil
 	}
 
+	template := cached.getTemplate()
+
+	// remove the template from all active drivers
+	// to ensure cleanup in case of a botched update
+	for driverN := range cached.activeDrivers {
+		driver, ok := c.drivers[driverN]
+		if !ok {
+			return resp, fmt.Errorf("%w: could not clean up %q", clienterrors.ErrNoDriver, driverN)
+		}
+
+		err := driver.RemoveTemplate(ctx, template)
+		if err != nil {
+			return resp, err
```

Comment:
return early here prevents it from removing from other drivers. is this intentional?

Reply:
Yep, the idea is that a failed removal needs to be met with a retry (as in a controller). If we cannot fully remove a template from a driver and cannot recover from that state, then the system is in an unknown state and the program should die (e.g. we have stale Rego code in OPA, who knows what impact it may have)